src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+        # unless defs is overriden.
+        if defs is None:
+            defs = self.__class__._defaults
+
+        finals = self.__class__._finalProperties
+
+        # Check for incompatible specifier combinations
+        specifier_names = [spec.name for spec in specifiers]
+
+        if "On" in specifier_names:
+            if collections.Counter(specifier_names)["On"] > 1:
+                raise RuntimeParseError(f'Cannot use "On" specifier to modify "On" specifier')
+
+        # Split the specifiers into two groups, normal and modifying. Normal specifiers set all relevant properties
+        # first. Then modifying specifiers can modify or set additional properties
+        normal_specifiers = [spec for spec in specifiers if not isinstance(spec, ModifyingSpecifier)]
+        modifying_specifiers = [spec for spec in specifiers if isinstance(spec, ModifyingSpecifier)]
+
+        '''
+        For each property specified by a normal specifier:
+            - If not in properties specified, properties[p] = specifier
+            - Otherwise, if property specified, check if specifier's priority is higher. If so, replace it with specifier
+
+        Priorties are inversed: A lower priority number means semantically that it has a higher priority level
+        '''
+        for spec in normal_specifiers:
+            assert isinstance(spec, Specifier), (name, spec)
+
+            # Iterate over each property.
+            for prop in spec.priorities:
+                # Check if this is a final property has been specified. If so, throw an assertion or error,
+                # depending on whether or not this object is internal.
+                if prop in finals:
+                    raise RuntimeParseError(f'property "{prop}" cannot be directly specified')
+
+
+                if prop in properties:
+                    # This property already exists. Check that it has not already been specified
+                    # at equal priority level. Then if it was previously specified at a lower priority
+                    # level, override it with the value that this specifier sets.
+                    if spec.priorities[prop] == priorities[prop]:
+                        raise RuntimeParseError(f'property "{prop}" specified twice with the same priority')
+                    if spec.priorities[prop] < priorities[prop]:
+                        properties[prop] = spec
+                        priorities[prop] = spec.priorities[prop]
+                else:
+                    # This property has not already been specified, so we should initialize it.
+                    properties[prop] = spec
+                    priorities[prop] = spec.priorities[prop]
+
+        '''
+        If a modifying specifier specifies the property with a higher priority,
+        set the object's property to be specified by the modifying specifier. Otherwise,
+        if the property exists and the priorities match, object needs to be specified
```

Comment:
Is "and the priorities match" accurate? The code below seems to allow modification if the modifying specifier uses a priority equal to **or lower** than the non-modifying specifier.